CorsixTH/Lua/entities/humanoids/staff/doctor.lua

```diff
@@ -249,17 +249,21 @@ local profile_attributes = {
   Psychiatrist = "is_psychiatrist",
   Surgeon = "is_surgeon",
   Researcher = "is_researcher",
+  Junior = "is_junior",
+  Consultant = "is_consultant",
 }
 
 -- Helper function to decide if Staff fulfills a criterion
--- (one of "Doctor", "Nurse", "Psychiatrist", "Surgeon", "Researcher" and "Handyman")
+-- (one of "Doctor", "Nurse", "Psychiatrist", "Surgeon", "Researcher" and "Handyman", "Receptionist", "Junior", "Consultant")
 function Doctor:fulfillsCriterion(criterion)
   if criterion == "Doctor" then
     return true
   elseif criterion == "Psychiatrist" or criterion == "Surgeon" or criterion == "Researcher" then
     if self.profile and self.profile[profile_attributes[criterion]] == 1.0 then
       return true
     end
+  elseif criterion == "Junior" or criterion == "Consultant" then
```

Comment:
This is the same logic as the condition above, but written differently which makes the function harder to read.

This staff is not a Nurse or Handyman so they wouldn't have those profile attributes.

Couldn't the whole thing be reduced to:
```
if criterion == "Doctor" then
  return true
end
if self.profile and self.profile[profile_attributes[criterion]] then
  return true
end
return false
```
To me that's  easier to read.

Reply:
Indeed, I took the countStaffOfCategory as boolean checks and just mapped them over, but really for is_consultant and is_junior it was 1 or nil (an existence check) and they do fall into that so can be simplified, just with the value check still, rather than just an existence check otherwise partially trained doctors will meet the skill.

Reply:
Good catch. Please do.